### PR TITLE
Using wmi

### PR DESCRIPTION
**Pull request by madchills**
This changes how the  beacon configures the folders for auditing.  Before it used powershell command set-acl which does not work on c:\Users and most importantly c:\windows.  I instead had to use WMI, which does not have that restriction.  With that I also found i need to change the remove ACE function to use WMI as well, since you also cannot remove ACEs from c:\windows using native powershell commands.
I also fixed many bugs. Most notably the bug that was using the parent path of the exclude instead of the exclude path.  Example: auditing c:\windows but excluding c:\windows\system32, would actually remove the ACEs from c:\windows, instead of the system32 folder.  Other minor bug fixes as well

### CODE REVIEW

File: _beacons/win_notify.py [hunk 1]
```diff
@@ -171,13 +172,19 @@ def beacon(config):
                     confirm = _add_acl(path, mask, wtype, recurse)
                     sys_check = 1
                 if config[path].get('exclude', False):
-                    _remove_acl(path)
+                    for item in config[path]['exclude']:
+                        if '*' in exclude:
+                            for wildcard_item in glob.iglob(item):
+                                _remove_acl(wildcard_item)
+                        _remove_acl(item)
```
Comment (basepi) [inline _beacons/win_notify.py]: Should this be under an `else`? You probably don't want to call `_remove_acl` on the path `C:\users\*\desktop`, right?

**Comment by basepi** [comment]
Left one comment inline. Otherwise looks great.

**Comment by madchills** [comment]
The _remove_acl function is just removing the ACE flag that marks the folder for monitoring.  If for some reason someone wants to monitor `c:\users` and all subfolders, but not `c:\users\*\desktop` and subfolders, this gives them that option.

File: _beacons/win_notify.py [hunk 2]
```diff
@@ -171,13 +172,19 @@ def beacon(config):
                     confirm = _add_acl(path, mask, wtype, recurse)
                     sys_check = 1
                 if config[path].get('exclude', False):
-                    _remove_acl(path)
+                    for item in config[path]['exclude']:
```
Comment (basepi) [inline _beacons/win_notify.py]: Should this be `for exclude in config[path]['exclude']:`?

File: _beacons/win_notify.py (hunk 1 above)
Comment (basepi) [inline _beacons/win_notify.py]: > The _remove_acl function is just removing the ACE flag that marks the folder for monitoring. If for some reason someone wants to monitor c:\users and all subfolders, but not c:\users*\desktop and subfolders, this gives them that option.

I understand that, but `c:\users\*\desktop` isn't a valid path, so I figure you should do `if '*' in exclude, do one thing, else, just remove the acl`. Does that make sense?

**Comment by madchills** [comment]
Good catch on both of those.  Fixed in e713db20b2736d8426fc12769e602a1a000b57c6 diff

File: _beacons/win_notify.py [hunk 3]
```diff
@@ -171,13 +172,20 @@ def beacon(config):
                     confirm = _add_acl(path, mask, wtype, recurse)
                     sys_check = 1
                 if config[path].get('exclude', False):
-                    _remove_acl(path)
+                    for exclude in config[path]['exclude']:
+                        if '*' in exclude:
+                            for wildcard_item in glob.iglob(item):
```
Comment (basepi) [inline _beacons/win_notify.py]:
```
for wildcard_item in glob.iglob(exclude):
```

File: _beacons/win_notify.py [hunk 4]
```diff
@@ -171,13 +172,20 @@ def beacon(config):
...
+                                _remove_acl(wildcard_item)
+                        else:
+                            _remove_acl(item)
```
Comment (basepi) [inline _beacons/win_notify.py]:
```
_remove_acl(exclude)
```

**Pull request merged**